src/container/oid_iv.c

```diff
@@ -331,6 +342,7 @@ oid_iv_reserve(void *ns, uuid_t po_uuid, uuid_t co_uuid, uint64_t num_oids, d_sg
 	oids->num_oids = num_oids;
 	oids->req_rank     = dss_self_rank();
 	oids->req_num_oids = num_oids;
+	oids->req_ptr      = value;
```

Comment:
the 'value' passed here is from https://github.com/daos-stack/daos/blob/0be2d10c2917257b3d1ca6d8af9fdfc0891d9aa2/src/container/srv_target.c#L2357 and it is an addr of a variable on a stack.

as such req_ptr can end up be the same for different iv values, which is not what we want.
i think instead oid->req_ptr should store 'oids' as the iov_buf will be different for different iv values and combined with req_rank should uniquely identify the 'iv transaction'

Reply:
this one is synchronous.. you cannot reuse a stack variable unless the call exits, right?

Reply:
well, not sure what you mean by synchronous here. 
If i am understanding this whole thing correctly:
you are storing it here in oid (iv value contents) and then those contents will possibly get shipped around to different engine and examined asynchronously there (e.g. in oid_iv_end_update() on that engine)

some intermediate engine can then get 2 oid_iv_end_update() calls for different iv values from the same rank, but those iv value contents will have same oid->req_rank and same oid->req_ptr (as req_ptr will both contain &sgl on a stack on when it was created on that remote engine). 

Where as if you stored "sg_iovs[0].iov_buf" during reserve then it would be unique on that rank for each iv value that is being worked on.

Reply:
After discussing over slack with @mchaarawi  I understand now that the remote stack for iv update will remain intact ensuring another iv_update happening from same rank ending up using a different value. 
LGTM as such.